frontend/packages/core/src/Input/select.tsx

```diff
@@ -241,8 +245,10 @@ export const Select = ({
     if (!value) {
       return;
     }
-    setSelectedIdx(flatOptions.findIndex(opt => opt?.value === value || opt?.label === value));
-    onChange && onChange(value);
+    const findIndex = val => flatOptions.findIndex(opt => opt.value === val || opt.label === val);
+    setSelectedIdxs(multiple ? value.map(val => findIndex(val)) : [findIndex(value)]);
+
+    onChange && onChange(selectedIdxs.join(","));
```

Comment:
I think this is the problem, we're returning the selectedIdxs joined and not their actual values like we do above

```suggestion
    onChange && onChange((selectedValues() ?? []).join(","));
```

Reply:
yeah thats def it lol. Let me add a breaking test and we can re-review after the fix

Reply:
Sounds good!